### DISCUSSION TITLE
How does updateDocument work while offline on Web (updateDoc vs await)?

### DISCUSSION THREAD

**Discussion by morzel85** [capawesome-team/capacitor-firebase#756] [Q&A]
The documentation for Firestore JS (Web) API for [_updateDoc_](https://firebase.google.com/docs/reference/js/firestore_.md#updatedoc_51a65e3) states that promise will not be resolved in offline mode (that's why we [usually don't await](https://github.com/firebase/firebase-js-sdk/issues/6515#issuecomment-1209836441)).

So I'm curious how does [_updateDocument_](https://github.com/capawesome-team/capacitor-firebase/blob/b57f1ee264f0a614e7284307402cf2293b665913/packages/firestore/src/web.ts#L124) function from  _capacitor-firebase_ work internally as it seems to call the _updateDoc_ with _await_, yet doesn't hang even if I put my browser in offline mode:

![image](https://github.com/user-attachments/assets/10d66c26-6a01-455e-be60-365666d0c909)

BTW: is awaiting on a call _FirebaseFirestore.updateDocument_ necessary/recommended?

**Comment by robingenz** [comment]
TBH, I wasn't aware of that. However, I'm glad that it works but I can't tell you why it works. 😅 We're just calling the Firebase JS SDK.

> BTW: is awaiting on a call FirebaseFirestore.updateDocument necessary/recommended?

That's up to you and your requirements. There is no recommendation.
